core/src/services/icloud/session.rs

```diff
@@ -0,0 +1,304 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::collections::{BTreeMap, HashMap};
+use std::fmt::{Debug, Formatter};
+
+use http::header;
+use http::Method;
+use http::Request;
+use http::Response;
+use http::StatusCode;
+
+use crate::types::Result;
+use crate::{Error, ErrorKind};
+
+use crate::raw::oio::WriteBuf;
+use crate::raw::{new_json_deserialize_error, AsyncBody, HttpClient, IncomingAsyncBody};
+use crate::services::icloud::error::parse_error;
+use crate::services::icloud::webservices::iCloudWebservicesResponse;
+use serde_json::json;
+
+//TODO It just support cn
+//Other country shoule be https://www.icloud.com
+//Need to improve?
+const GLOBAL_HEADERS: [(&str, &str); 2] = [
+    ("Origin", "https://www.icloud.com.cn"),
+    ("Referer", "https://www.icloud.com.cn/"),
+];
+
+static ACCOUNT_COUNTRY_HEADER: &str = "X-Apple-ID-Account-Country";
+static OAUTH_STATE_HEADER: &str = "X-Apple-OAuth-State";
+
+static SESSION_ID_HEADER: &str = "X-Apple-ID-Session-Id";
+
+static SCNT_HEADER: &str = "scnt";
+
+static SESSION_TOKEN_HEADER: &str = "X-Apple-Session-Token";
+static AUTH_ENDPOINT: &str = "https://idmsa.apple.com/appleauth/auth";
+static SETUP_ENDPOINT: &str = "https://setup.icloud.com/setup/ws/1";
+
+static APPLE_RESPONSE_HEADER: &str = "X-Apple-I-Rscd";
+
+const AUTH_HEADERS: [(&str, &str); 7] = [
+    (
+        "X-Apple-OAuth-Client-Id",
+        "d39ba9916b7251055b22c7f910e2ea796ee65e98b2ddecea8f5dde8d9d1a815d",
```

Comment:
Where does this key come from?

Reply:
ref:https://github.com/picklepete/pyicloud/blob/master/pyicloud/base.py#L392C1-L392C1

Reply:
Please add in comments so the maintainer knows where it's from.